site/content/contribute/more-info/webapp/developer-workflow.md

```diff
@@ -1,45 +1,78 @@
+
 9. That's it! Rejoice that you've helped make Mattermost better.
 
+### Package structure
+
+The web app is set up as a monorepo which has the code broken up into multiple packages. The main packages in the web app are:
+
+* `boards` - The Boards section of the app. It was previously located at https://github.com/mattermost/focalboard/.
+* `channels` - The main web app which contains Channels, the System Console, login/signup pages, and most of the core infrastructure for the app. It was previously located at https://github.com/mattermost/mattermost-webapp/.
+* `platform` - Packages used by the web app and related projects
+    * `client` - The JavaScript client for Mattermost's REST API, available on NPM as {{< newtabref href="https://www.npmjs.com/package/@mattermost/client" title="@mattermost/client" >}}
+    * `components` - A work-in-progress package containing UI components designed to be used by different parts of Mattermost
+    * `types` - The TypeScript types used by Mattermost, available on NPM as {{< newtabref href="https://www.npmjs.com/package/@mattermost/types" title="@mattermost/types" >}}
+* `playbooks` - The Playbooks section of the app. It was previously located at https://github.com/mattermost/mattermost-plugin-playbooks/.
+
+These packages will likely change over time as we aim to better integrate Boards, Channels, and Playbooks while also pulling out shared code into separate packages for easier sharing and maintainability. For now though, `channels` contains most of the web app code, so it's likely the first place you'll want to go if you're new to the project.
+
+Each package has a slightly different structure, but each package generally contains a `src` directory with subfolders for React components, Redux actions, i18n files and so on. As an example, `channels` contains:
+
+* `actions` - Contains Redux actions which make up much of the view logic for the web app
+* `components` - Contains UI components nad views written using React
+* `i18n` - Contains the localization files for the web app
+* `packages/mattermost-redux` - Contains most of the Redux logic used for handling data from the server
+* `plugins` - Contains the plugin framework, utility functions and components
+* `reducers` - Contains Redux reducers used for view state
+* `selectors` - Contains Redux selectors used for view state
+* `tests` - Contains setup code and mocks used for unit testing
+* `utils` - Contains many widely-used utility functions
+
+### Important libraries and technologies
```

Comment:
I forget why I decided to add this section calling out libraries, but it seems helpful. I might expand this in the future with build tools like Webpack or TypeScript

Reply:
Yup definitely like MUI and such